fbw-a32nx/src/systems/instruments/src/PFD/HeadingIndicator.tsx

```diff
@@ -207,27 +207,49 @@ interface GroundTrackBugProps {
 }
 
 class GroundTrackBug extends DisplayComponent<GroundTrackBugProps> {
-    private trackIndicator = FSComponent.createRef<SVGGElement>();
+    private groundTrack = Arinc429ConsumerSubject.create(null);
+
+    private readonly style = ObjectSubject.create({
+        display: '',
+        transform: 'translate3d(0px, 0px, 0px)',
+    });
+
+    private isVisibleSub = MappedSubject.create(([groundTrack, heading]) => {
+        const delta = getSmallestAngle(groundTrack.value, heading);
+        // TODO should also be hidden if heading is invalid
+        return groundTrack.isNormalOperation() && Math.abs(delta) < DisplayRange;
+    }, this.groundTrack, this.props.heading);
+
+    private transformSub = MappedSubject.create(([groundTrack, heading]) => {
+        const offset = getSmallestAngle(groundTrack.value, heading) * DistanceSpacing / ValueSpacing;
+        return `translate3d(${offset}px, 0px, 0px)`;
+    }, this.groundTrack, this.props.heading);
 
     onAfterRender(node: VNode): void {
         super.onAfterRender(node);
 
-        const sub = this.props.bus.getSubscriber<DmcLogicEvents>();
+        const sub = this.props.bus.getArincSubscriber<DmcLogicEvents>();
 
-        sub.on('track').handle((groundTrack) => {
-            if (groundTrack.isNormalOperation()) {
-                const offset = getSmallestAngle(groundTrack.value, this.props.heading.get()) * DistanceSpacing / ValueSpacing;
-                this.trackIndicator.instance.style.display = 'inline';
-                this.trackIndicator.instance.style.transform = `translate3d(${offset}px, 0px, 0px)`;
+        this.groundTrack.setConsumer(sub.on('track').withArinc429Precision(3));
+
+        this.isVisibleSub.sub((isVisible) => {
+            if (isVisible) {
+                this.style.set({ display: '' });
+                this.transformSub.resume();
             } else {
-                this.trackIndicator.instance.style.display = 'none';
+                this.style.set({ display: 'none' });
```

Comment:
You could set the style object inline in the JSX with subscribables for each prop to save allocating a new object each time.

Reply:
Sounds good, will apply